### PR TITLE
fix behavior + test

### PR DESCRIPTION
**Pull request by roninsightrx**
This fixes the edge case where it was possible to have duplicate `obs_type` columns in the return object from `parse_input_data()`. 

It's still a bit ugly to have a specific line of code to set OBS_TYPE to NULL to avoid an edge case, but I think this is the simplest solution. The other solution is to not care about the edge case (it essentially only occurs with erroneous usage).

Very minor change, shouldn't affect any production code.

### CODE REVIEW

File: R/parse_input_data.R [hunk 1]
```diff
@@ -13,6 +13,7 @@ parse_input_data <- function(
   } else {
     data$obs_type <- data[[obs_type_label]]
   }
+  data$OBS_TYPE <- NULL # ensure there is no duplicate `obs_type` column
```
Comment (mccarthy-m-g) [inline R/parse_input_data.R:16]: GitHub doesn't let me make suggestions on unchanged lines. But instead of this, you could do just move `colnames(data) <- tolower(colnames(data))` to the start of the function, then make sure `obs_type_label` is also lowercase:

```r
parse_input_data <- function(
  data,
  cols = list(x = "t", y = "y"),
  obs_type_label = NULL
) {
  colnames(data) <- tolower(colnames(data)) # HERE
  ## Parse data to include label for obs_type
  if(is.null(obs_type_label)) {
    data$obs_type <- 1
  } else {
    data$obs_type <- data[[tolower(obs_type_label)]] # HERE
  }
  if(!all(tolower(unlist(cols)) %in% names(data))) {
    stop("Expected column names were not found in data. Please use 'cols' argument to specify column names for independent and dependent variable.")
  }
  if(inherits(data, "PKPDsim")) {
    # when fitting directly from PKPDsim-generated data:
    if("comp" %in% names(data)) {
      data <- data[data$comp == "obs",]
      data$evid <- 0
    }
  }
  if("evid" %in% colnames(data)) {
    data <- data[data$evid == 0,]
  }
  data[order(data$t, data$obs_type),]
}
```

All tests pass with this change, but you'll need to change the expected order for your new tests from `c("t", "y", "obs_type")` to `c("t", "obs_type", "y")`.

File: tests/testthat/test-parse_input_data.R [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
     OBS_TYPE = c(2, 1, 1),
     Y = c(10, 20, 30)
   )
   
-  result <- parse_input_data(data)
+  result <- parse_input_data(data, obs_type_label = NULL)
   
   # Check column names are lowercase
-  expect_equal(names(result), c("t", "obs_type", "y", "obs_type"))
+  expect_equal(names(result), c("t", "y", "obs_type"))
```
Comment (mccarthy-m-g) [inline tests/testthat/test-parse_input_data.R:12]:
```suggestion
  expect_equal(names(result), c("t", "obs_type", "y"))
```

File: tests/testthat/test-parse_input_data.R [hunk 3]
```diff
@@ -1,21 +1,42 @@
...
+  data <- data.frame(
+    T = c(2, 1, 3),
+    OBS_TYPE = c(2, 1, 1),
+    Y = c(10, 20, 30)
+  )
+  
+  result <- parse_input_data(data, obs_type_label = "OBS_TYPE")
+  
+  # Check column names are lowercase
+  expect_equal(names(result), c("t", "y", "obs_type"))
```
Comment (mccarthy-m-g) [inline tests/testthat/test-parse_input_data.R:43]:
```suggestion
  expect_equal(names(result), c("t", "obs_type", "y"))
```

**Review by mccarthy-m-g — COMMENTED** [review]
Would this approach also work? It's a bit cleaner and more general.

File: R/parse_input_data.R (hunk 1 above)
Comment (roninsightrx) [reply R/parse_input_data.R:16]: On first impression this looks indeed cleaner. However, I think the one drawback is that it creates another, admittedly even more esoteric, edge case: when a user has e.g. both "obs_type" and "OBS_TYPE" as columns, and wants to use either one... It may not throw an error but it becomes unpredictable which one it will pick, or perhaps we still end up with two columns in that case? (Not tested)

Comment (roninsightrx) [reply R/parse_input_data.R:16]: Actually, I don't even know if my version works for that case :) Let me try that tonight

Comment (roninsightrx) [reply R/parse_input_data.R:16]: Yeah, I think better to stick with the original solution. With the above update it is still possible to have two columns obs_type in the output. I've added a test case for that.

**Review by mccarthy-m-g — APPROVED** [review]
lgtm!
